cfme/infrastructure/datastore.py

```diff
@@ -151,20 +159,17 @@ def get_vms(self):
 
     def _on_vms_page(self):
         """ Returns ``True`` if on the datastore vms page, ``False`` if not."""
-        return sel.is_displayed(
-            '//div[@class="dhtmlxInfoBarLabel-2"][contains(., "{}") and contains(., "{}")]'.format(
-                self.name, "All Registered vms")
-        )
+        return summary_title() == '{} ({})'.format(self.name, "All Registered VMs")
 
     def delete_all_attached_vms(self):
-        sel.force_navigate('infrastructure_datastore', context=self._get_context())
```

Comment:
And these are why you probably needed to add the ```ensure_browser_open()```. The force_navigate is a natural first step to any publicly available API call.